### PR TITLE
Show a user's total of stars received

### PR DESCRIPTION
**Pull request by momothereal**
Shows the total of stars received on a user's profile.

Example:

image

### CODE REVIEW

File: source/features/show-user-stars-received.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+	let page = 1;
+	for (;;) {
+		// eslint-disable-next-line no-await-in-loop
+		const repoList = await fetchApi(`users/${username}/repos?type=owner&page=${page}&per_page=100`).catch(() => null);
+		console.log(repoList);
+		if (!repoList || repoList.length === 0) {
+			break;
+		}
+		for (const repoObj of repoList) {
+			if (Object.prototype.hasOwnProperty.call(repoObj, 'stargazers_count')) {
```
Comment (sindresorhus) [inline source/features/show-user-stars-received.js]: Use `Reflect.has`

File: source/features/show-user-stars-received.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+	if (starCount >= 1000) {
+		// Simplify
+		starCount = (Math.round(starCount / 100) / 10) + 'k';
+	}
+
+	detailsContainer.append(
+		<li itemprop={'starCount'} itemProple={'false'} aria-label={'Star Count'}
+			className={'vcard-detail pt-1 css-truncate css-truncate-target'}>
+			{icons.star()}
+			<span className={'p-label'}>Stars Received: {starCount}</span>
```
Comment (sindresorhus) [inline source/features/show-user-stars-received.js]: I would put it like this: `300 stars received`

**Comment by sindresorhus** [comment]
I'm not a big fan of vanity metrics like this, but I guess many would want this, so why not. However, it's way too slow. It took 10 seconds to load on my profile. By then, people would have gone elsewhere. Maybe look into using the GraphQL API instead, where you can just extract exactly the information you need.

**Comment by momothereal** [comment]
One of the issues I faced when implementing this is pagination; it loads each page (of 100 repositories max) until none is returned. However, it would be possible to get the number of pages from the public repository count of a user. From there, the repo lists can be fetched asynchronously.

If that doesn't make it much faster, I'll try to find other ways, like the GQL API you mentioned.

**Comment by sindresorhus** [comment]
I think you should try GraphQL right away. It could potentially let us compute the count server-side, so all we have to do is to fetch the count.

**Comment by sindresorhus** [comment]
You can borrow some GraphQL request code from here: https://github.com/sindresorhus/refined-github/pull/1399/files#diff-1285b6ab4d237edd45e7c9b6e9412928

**Comment by momothereal** [comment]
I'm not aware of a sum aggregation feature in GraphQL, so it would still require fetching pages of 100 repos sequentially and summing the stargazers of each repo. It does have the advantage of only fetching the necessary field, however.

**Comment by sindresorhus** [comment]
Relevant GitHub GraphQL forum thread: https://platform.github.community/t/best-way-to-gather-all-stars-a-user-has-received/1068/2

**Comment by sindresorhus** [comment]
> It does have the advantage of only fetching the necessary field, however.

That's a pretty big advantage.

**Comment by momothereal** [comment]
I just tested with GraphQL. The performance is identical to the original implementation, mainly because it relies on a cursor (making the traverse synchronous). I don't know of a way to traverse pages asynchronously because of the cursor system.

So far, the most efficient solution is the asynchronous requests (current impl).

**Comment by momothereal** [comment]
@hkdobrev I might be wrong, but I think this is what the `Popular repositories` section does (when it's not overriden by the `Pinned repositories` feature).

I think it would be useful to add a toggle feature to show `Popular repositories` when the user has pinned repos. This would find the 6 repos with the most stars (using the GraphQL search).

**Comment by momothereal** [comment]
Discarding in favor of #1495

**Pull request closed without merging**